cg/server/endpoints/cases.py

```diff
@@ -16,30 +19,28 @@
 @CASES_BLUEPRINT.route("/cases")
 def get_cases():
     """Return cases with links for a customer from the database."""
-    enquiry: str = request.args.get("enquiry")
-    action: str = request.args.get("action")
+    cases_request = CasesRequest.model_validate(request.args.to_dict())
 
     customers: list[Customer] = _get_current_customers()
-    cases: list[Case] = _get_cases(enquiry=enquiry, action=action, customers=customers)
+    cases, total = _get_cases(request=cases_request, customers=customers)
 
-    nr_cases: int = len(cases)
     cases_with_links: list[dict] = [case.to_dict(links=True) for case in cases]
-    return jsonify(families=cases_with_links, total=nr_cases)
+    return jsonify(cases=cases_with_links, total=total)
 
 
 def _get_current_customers() -> list[Customer] | None:
     """Return customers if the current user is not an admin."""
     return g.current_user.customers if not g.current_user.is_admin else None
 
 
-def _get_cases(
-    enquiry: str | None, action: str | None, customers: list[Customer] | None
-) -> list[Case]:
+def _get_cases(request: CasesRequest, customers: list[Customer] | None) -> tuple[list[Case], int]:
     """Get cases based on the provided filters."""
     return db.get_cases_by_customers_action_and_case_search(
-        case_search=enquiry,
+        action=request.action,
+        case_search=request.enquiry,
         customers=customers,
-        action=action,
+        offset=request.page * request.page_size,
```

Comment:
depends on the starting index in the front end. Is teh first page 0 or 1 @moedarrah?
```suggestion
        offset=(request.page-1) * request.page_size,
```

Reply:
The initial page should be page: 1.